src/Network/AWS/ARN.hs

```diff
@@ -171,35 +169,40 @@ _ARN = prism' renderARN parseARN
 
 -- | Split a 'Text' into colon-separated parts.
 --
--- This is useful for editing the resource part of an ARN:
+-- This is an improper 'Iso'' (@Text.intercalate ":"@ is a retraction
+-- of @Text.splitOn ":"@ but not a section). The 'Iso'' laws are
```

Comment:
The terms "retraction" and "section" might be a bit jarring? This sentence is redundant with the following counterexamples anyway.

Reply:
Disagree with the redundancy - it precisely describes what goes wrong. How about: `@Text.intercalate ":" . Text.splitOn ":" = id`@, but @Text.splitOn ":" . Text.intercalate ":" /= id@`?

Reply:
I think that's better